strawberryfields/ops.py

```diff
@@ -391,9 +391,11 @@ def merge(self, other):
         # channels can be merged if they are the same class and share all the other parameters
         if self.p[1:] == other.p[1:]:
             # determine the combined first parameter
-            T = self.p[0] * other.p[0]
+
+            T = np.dot(other.p[0], self.p[0])
             # if one, replace with the identity
-            if T == 1:
+            T_arr = np.atleast_2d(T)
+            if np.allclose(T_arr, np.eye(T_arr.shape[0])):
```

Comment:
Not 100% sure myself, but this doesn't have any effect for users using Channels with the TF backend?

I assume not, since the tests still passed, although it could be something the test suite is missing :)

Reply:
When p[0] is a scalar (as is assumed in the previous version) I think the only operational difference after my change is that something within tolerance of 1 will be assumed to be identity, whereas previously it had to be exact. Is this okay? Is there an appropriate tol variable I can pass into the ``np.allclose``?